spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/reload/ConfigReloadAutoConfiguration.java

```diff
@@ -102,6 +102,12 @@ public ConfigurationChangeDetector propertyChangeWatcher(
 					"Unsupported configuration reload mode: " + properties.getMode());
 		}
 
+		@Bean
+		@ConditionalOnMissingBean
+		public TaskScheduler taskScheduler() {
```

Comment:
Creating a bean of type `TaskScheduler` was brought up in s-c-consul. https://github.com/spring-cloud/spring-cloud-consul/issues/723
We should come up with some consistency

Reply:
I guess my thought was to create a bean that the app could then customize if they wanted to.  I think the approach `spring-vault` takes makes it so there is no collision between `TaskScheduler` beans an app might create and the one used by `spring-vault`.  I could see both approaches being valid.  Maybe a nice middle ground is we use a `TaskSchedulerWrapper` but make it `@ConditionalOnMissingBean` so that if they do want to customize the behavior of that particular `TaskScheduler` they could but at the same time if the app creates its own `TaskScheduler` we avoid any kind of unintended consequences?

Reply:
That sounds like a good idea. I'll do this in that consul issue as well.